### DISCUSSION TITLE
FreeSewing Roadmap

### DISCUSSION THREAD

**Discussion by joostdecock** [freesewing/freesewing#1278] [Announcements]
## What am I looking at? 🤔

This is the FreeSewing roadmap. All the things that we would like to implement sooner or later are on this roadmap.
If you'd like to see where the project is heading, this should give you a good indication.

The roadmap is divided into different topics/sections and as such is not chronologic. In other words, just because something is listed first does not mean it will be implemented first.

> **Can I add something to the roadmap?**
> 
> You can leave a comment to make a suggestion. But we don't put things on the roadmap unless there's at least some commitment from somebody to do some actual work on it.

> **Note**: Some of the issues on the roadmap are closed. That's does not mean it is implemented, but rather that we want to keep the list of open issues to a manageable amount. So if something is added to the roadmap, we close the issue and reference it here for more info / backstory. 

## Documentation

- Document all options per design, explain the nature of how they are parametric

## Backend

- #4003 

## Frontends
- #6552 
- Support adding new designs to the stand-alone development environment. See #6397 
- Add support for multi-set patterns
- Provide a logical (to the body) sort order for measurements #4007 
- Add some sort of measurements-by-invite system
- Add support for exporting & importing People/Measurements #962 
- Support themeing of the SVG output. Not just by rolling your own plugin, but in a way that lets website users override styles.
This is needed for people using projectors, plotters, or other specific setups. Might be an idea to load styles from a URL/gist so that people can publish their own.

## Pattern design

- Support off-the-shoulder draft in Brian #203 
- Make simon & simone support button placement for left-handed people #463 
- Support wider hips than chest in Wahid #1361 
- Belly support in patterns #2531 

## Plugins

- Add a macro to rotate paths #406

## Already implemented

The following ideas have graduated from this very roadmap and made their way into our codebase:

- [x] Support *snapping* of percentage options
- [x] Migrate blog posts to Strapi
- [x] Migrate showcase posts to Strapi
- [x] Migrate newsletter to Strapi
- [x] Move markdown content into monorepo
- [x] Merge Crowdin translation projects
- [x] Add endpoint to backend for auto-generated open graph images
- [x] Improve display of contributor profiles
- [x] Migrate freesewing.dev from GatsbyJS to NextJS
- [x] Add scale setting to improve support for doll-sized patterns
- [x] Move backend code into monorepo
- [x] Provide a standard way for conditional display of options in the frontend
- [x] Add support for pan & zoom on pattern view
- [x] Allow manual input on slider options
- [x] Support for reporting the absolute value of options
- [x] Switch from rollup to esbuild
- [x] Move from CRA to a NextJS based developer environment next to the pattern code
- [x] Provide an online developer environment
- [x] Migrate style to TailwindCSS
- [x] Avoid babel
- [x] Make all measurements available in all models
- [x] Support for _packs_
- [x] Remove dependency on MaterialUI
- [x] ESM only, no more CJS
- [x] Only use named exports
- [x] Change the way sampling is handled
- [x]  Debug support / Allow overriding of log methods
- [x] Allow for multiple (sets of) settings in our draft
- [x] Migrate backend from mongodb to sqlite
- [x] Port PDF generation and tiling to Javascript
- [x] Migrate strapi to Sanity
- [x] Add support for manual pattern layout
- [x] Better open graph support
- [x] Migrate to NextJS
- [x] Move frontend code to monorepo
- [x] Provide better technical drawings for designs
- [x] Allow users to avoid all measurements comparisons
- [x] Implement bookmarks. For docs, but also for (curated) sets, designs, patterns.
- [x] Write *designer notes* for all designs
- [x] Add filter based on various tag types to the design view
- [x] Add bulge option to Uma #3695 
- [x] Allow users to submit showcases
- [x] Support a standard way to handle run-time decisions about how patterns should be drafted #203 
See #203 for an example
- [x] Improve pattern page layout.

**Comment by bekaness** [comment]
![image](https://user-images.githubusercontent.com/25408559/139040240-91c6af6d-7d1b-46fd-86c0-3e99c9a94124.png)
I am commenting as requested to be able to change stroke width in scaling patterns down to small sizes (such as for dolls.) Thank you for the opportunity to be a part of this community.

**Reply by joostdecock** [reply]
FYI: https://canary.freesewing.dev/reference/api/settings/scale

**Comment by bekaness** [comment]
Thank you so very much!!!!!!!!!!!!!!!!!

Sent with [ProtonMail](https://protonmail.com/) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by HaasJona** [comment]
Rotating a path has been implemented (as a simple method: https://freesewing.dev/reference/api/path/rotate - not a macro)

**Comment by bekaness** [comment]
Thank you!!!!
